src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/reactors/Reactor.java

```diff
@@ -299,6 +300,43 @@ public int getGeneratedOutput(Location l, Config data) {
         }
     }
 
+    @Override
+    public int peekGeneratedOutput(@Nonnull Location l, @Nonnull Config data) {
+        BlockMenu inv = BlockStorage.getInventory(l);
+        BlockMenu accessPort = getAccessPort(l);
+        FuelOperation operation = processor.getOperation(l);
+
+        if (operation != null) {
+            return peekGenerateEnergy(l, data, inv, accessPort, operation);
+        } else {
+            return 0;
+        }
+    }
+
+    private int peekGenerateEnergy(@Nonnull Location l, @Nonnull Config data, @Nonnull BlockMenu inv, @Nullable BlockMenu accessPort, FuelOperation operation) {
```

Comment:
Can we extract some of the dupe code here? There's no reason to do a bunch of this in both peekGenerateEnergy and generateEnergy